integration-cli/docker_cli_run_test.go

```diff
@@ -2912,3 +2912,45 @@ func TestRunAllowPortRangeThroughPublish(t *testing.T) {
 	}
 	logDone("run - allow port range through --expose flag")
 }
+
+func TestRunOOMExitCode(t *testing.T) {
+	defer deleteAllContainers()
+
+	oomSource := `#include <stdlib.h>
+int main()
+{
+	int size = 1024 * 1024 * sizeof(char);
+	while (1) {
+		if (!(int *)calloc(1, size)) {
+			exit(1);
+		}
+	}
+}`
+
+	f, err := ioutil.TempFile("", "oom_")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f.Close()
+	gccCmd := exec.Command("gcc", "-static", "-x", "c", "-o", f.Name(), "-")
+	stdin, err := gccCmd.StdinPipe()
```

Comment:
I'm not sure, but probably `gccCmd.Stdin = bytes.NewBufferString(oomSource)` can be simpler. I'm okay with this, but just FYI.